src/lib/components/globalhealthbar/utils.tsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import { RectRadius } from 'recharts/types/shape/Rectangle';
+import {
+  DATE_FORMATER,
+  TIME_FORMATER,
+  TIME_SECOND_FORMATER,
+} from '../date/FormattedDateTime';
+
+const oneHour = 60 * 60 * 1000;
+const oneDay = 24 * oneHour;
+
+export const setHistoryTooltipPosition = (
+  startDate: number,
+  endDate: number,
+  selectedDate: number | undefined,
+): string | null => {
+  if (selectedDate) {
+    const width = ((selectedDate - startDate) / (endDate - startDate)) * 600;
+    const leftPosition = width - 132 / 2;
+    return `auto auto -4px ${leftPosition}px`;
+  }
+  return 'auto';
+};
+
+export const getRadius = (
+  start: number,
+  end: number,
+  startDate: number,
+  endDate: number,
```

Comment:
What the meaning of those props ? IMO the signature of the function is not very clear to understand what it does